src/utils/tbinary_pool.go

```diff
@@ -0,0 +1,87 @@
+package utils
```

Comment:
Nit: by convention let's maybe put this in: `src/x/thrift` instead of `utils`. Also, add a package comment for it explaining where this came from? Maybe link back to the PR.